ui-tests-protractor/e2e/common/common.steps.ts

```diff
@@ -195,13 +195,45 @@ class CommonSteps {
   }
 
   @when(/^she selects "([^"]*)" from "([^"]*)" dropdown$/)
-  public async selectFromDropDown(option: string, selectId: string): P<any> {
+  public async selectFromDropDownById(option: string, selectId: string): P<any> {
     const selectSelector = `select[id="${selectId}"]`;
     const selectElement = await this.world.app.getElementByCssSelector(selectSelector);
     await browser.wait(ExpectedConditions.visibilityOf(selectElement), 6000, `Input ${selectId} not loaded in time`);
     return this.world.app.selectOption(selectElement, option);
   }
 
+  @when(/^she selects "([^"]*)" from "([^"]*)" selector-dropdown$/)
+  public async selectFromDropDownByElement(option: string, selectorName: string): P<any> {
+
+    let selectSelector;
+
+    switch (selectorName) {
```

Comment:
I'm not sure if this is best way, it would be better to make some class for getting selectors and keep selectFromDropDownMenu general as possible. Usage of method written this way could be quite limited.